pymc/distributions/timeseries.py

```diff
@@ -32,6 +36,191 @@
     "MvStudentTRandomWalk",
 ]
 
+from pymc.util import check_dist_not_registered
+
+
+class GaussianRandomWalkRV(RandomVariable):
+    """
+    GaussianRandomWalk Random Variable
+    """
+
+    name = "GaussianRandomWalk"
+    ndim_supp = 1
+    ndims_params = [0, 0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("GaussianRandomWalk", "\\operatorname{GaussianRandomWalk}")
+
+    def _supp_shape_from_params(self, dist_params, reop_param_idx=0, param_shapes=None):
+        steps = dist_params[3]
+
+        return (steps + 1,)
+
+    @classmethod
+    def rng_fn(
+        cls,
+        rng: np.random.RandomState,
+        mu: Union[np.ndarray, float],
+        sigma: Union[np.ndarray, float],
+        init: float,
+        steps: int,
+        size: Tuple[int],
+    ) -> np.ndarray:
+        """Gaussian Random Walk generator.
+
+        The init value is drawn from the Normal distribution with the same sigma as the
+        innovations.
+
+        Notes
+        -----
+        Currently does not support custom init distribution
+
+        Parameters
+        ----------
+        rng: np.random.RandomState
+           Numpy random number generator
+        mu: array_like
+           Random walk mean
+        sigma: np.ndarray
+            Standard deviation of innovation (sigma > 0)
+        init: float
+            Initialization value for GaussianRandomWalk
+        steps: int
+            Length of random walk, must be greater than 1. Returned array will be of size+1 to
+            account as first value is initial value
+        size: int
+            The number of Random Walk time series generated
+
+        Returns
+        -------
+        ndarray
+        """
+
+        if steps is None or steps < 1:
+            raise ValueError("Steps must be None or greater than 0")
```

Comment:
I think we can allow steps to be zero, in which case the distribution is just init

Reply:
Im inclined not to do this because i dont think anyone would want to do this on purpose and if they do it accidentally this will help them catch the mistake

Reply:
I disagree. Even for educational purposes it might be useful to show such a progression from init to first steps. 

We can make steps a required argument (I don't remember why we reverted on that decision) which will force you to choose steps. That's probably safer than our current default of 1, specially since its a new argument.

Reply:
Ok i'll open an issue to track for another PR so we can get this one merged